### PR TITLE
Updates to gene/affecting_entity_phenotype service to work with relat…

### PR DESCRIPTION
**Pull request by Shalsh23**
…ion graphs

### CODE REVIEW

File: src/main/scala/org/phenoscape/kb/queries/GeneAffectingPhenotype.scala [hunk 1]
```diff
@@ -147,11 +145,11 @@ object GeneAffectingPhenotype {
     if (entity.nonEmpty || quality.nonEmpty) {
       val entityPattern = entity
         .map { e =>
-          if (parts) sparql"?phenotype $rdfType/$PhenotypeOfSome/$rdfsSubClassOf/$PartOfSome $e . "
-          else sparql"?phenotype $rdfType/$PhenotypeOfSome $e . "
+          if (parts) sparql"?phenotype $phenotype_of/$part_of $e . "
```
Comment (balhoff) [inline src/main/scala/org/phenoscape/kb/queries/GeneAffectingPhenotype.scala:148]: Probably should be `$phenotype_of|($phenotype_of/$part_of)`

**Review by balhoff — CHANGES_REQUESTED** [review]

**Comment by Shalsh23** [comment]
Test 1:
iri: `http://purl.obolibrary.org/obo/UBERON_0008897`
quality: _
parts: `false`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

Old KB:
```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 504
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":0}
```

**Comment by Shalsh23** [comment]
Test 2:
iri: `http://purl.obolibrary.org/obo/UBERON_0000970`
quality: _
parts: `false`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000970&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 4813
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000970&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":0}
```

**Comment by balhoff** [comment]
I pushed a commit that is at least a partial fix. Needs more testing.

**Comment by Shalsh23** [comment]
> Test 1:
> iri: `http://purl.obolibrary.org/obo/UBERON_0008897`
> quality: _
> parts: `false`
> historical_homologs: `false`
> serial_homologs: `false`
> limit: `20`
> offset: `0`
> total: `true`
> 
> Old KB:
> 
> ```
> curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {
>   "total": 504
> }
> ```
> 
> New KB:
> 
> ```
> curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {"total":0}
> ```

Test 1 results after edits:

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":315}
```

**Comment by Shalsh23** [comment]
> Test 2:
> iri: `http://purl.obolibrary.org/obo/UBERON_0000970`
> quality: _
> parts: `false`
> historical_homologs: `false`
> serial_homologs: `false`
> limit: `20`
> offset: `0`
> total: `true`
> 
> ```
> curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000970&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {
>   "total": 4813
> }
> ```
> 
> New KB:
> 
> ```
> curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000970&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {"total":0}
> ```

Test 2 results after edits:

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0000970&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":3363}
```

**Comment by Shalsh23** [comment]
Test 3:

iri: `http://purl.obolibrary.org/obo/UBERON_0001793`
quality: `http://purl.obolibrary.org/obo/PATO_0001995`
parts: `false`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

Old KB:
```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001995&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 6
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001995&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":0}
```

**Comment by Shalsh23** [comment]
> http://purl.obolibrary.org/obo/PATO_0001995

Need to run the above tests with `http://www.ontobee.org/ontology/PATO?iri=http://purl.obolibrary.org/obo/PATO_0000001` and see if we get results.

**Comment by Shalsh23** [comment]
> Test 3:
> 
> iri: `http://purl.obolibrary.org/obo/UBERON_0001793`
> quality: `http://purl.obolibrary.org/obo/PATO_0001995`
> parts: `false`
> historical_homologs: `false`
> serial_homologs: `false`
> limit: `20`
> offset: `0`
> total: `true`
> 
> Old KB:
> 
> ```
> curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001995&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {
>   "total": 6
> }
> ```
> 
> New KB:
> 
> ```
> curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001995&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
> ```
> 
> ```
> {"total":0}
> ```

Test 4:

iri: `http://purl.obolibrary.org/obo/UBERON_0001793`
quality: ` http://purl.obolibrary.org/obo/PATO_0000001`
parts: `false`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

Old KB:
```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000001&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 12
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001793&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000001&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":16}
```

**Comment by Shalsh23** [comment]
Test 5:

Test 4:

iri: `http://purl.obolibrary.org/obo/UBERON_0008897`
quality: `http://purl.obolibrary.org/obo/PATO_0000052`
parts: `false`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

Old KB:
```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000052&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 283
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000052&parts=false&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":120}
```

**Comment by Shalsh23** [comment]
Test 6:

iri: `http://purl.obolibrary.org/obo/UBERON_0008897`
quality: `http://purl.obolibrary.org/obo/PATO_0000052`
parts: `true`
historical_homologs: `false`
serial_homologs: `false`
limit: `20`
offset: `0`
total: `true`

Old KB:
```
curl -X GET "https://kb.phenoscape.org/api/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000052&parts=true&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{
  "total": 311
}
```

New KB:
```
curl -X GET "http://localhost:8082/gene/affecting_entity_phenotype?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897&quality=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000052&parts=true&historical_homologs=false&serial_homologs=false&limit=20&offset=0&total=true" -H "accept: application/json"
```
```
{"total":131}
```

**Comment by balhoff** [comment]
The reduced results for E+Q queries are correct. This update fixes a bug that was returning too many results in the old code.

**Review by balhoff — APPROVED** [review]

**Pull request merged**